server/pulp/server/async/app.py

```diff
@@ -30,6 +31,50 @@
 _logger = logging.getLogger(__name__)
 
 
+class HeartbeatStep(bootsteps.StartStopStep):
+
+    requires = ('celery.worker.components:Timer', )
+
+    def __init__(self, consumer, **kwargs):
+        """
+        The step init is called when the Consumer instance is created, It is called with the
+        consumer instance as the first argument and all keyword arguments from the original
+        Consumer.__init__ call.
+        """
+        self.tref = None
+
+    def start(self, consumer):
+        """
+        This method is called when the worker starts up and also whenever the AMQP connection is
+        reset (which triggers an internal restart). The timer is reset when the connection is lost,
+        so we have to install the timer again for every call to self.start.
+        """
+        self.tref = consumer.timer.call_repeatedly(
+            30.0, self.record_heartbeat, (consumer, ), priority=10,
```

Comment:
Once PR #2907 goes through and you rebase onto it, you will want to replace the magic number "30" with the constant ```PULP_PROCESS_HEARTBEAT_INTERVAL``` from ```pulp.common.constants```.  FYI, that interval is 5 seconds, which is the new timing for it.  In case you need to change comments/docs.